### ISSUE TITLE
[TOTP] empty and same name entries don't error

### ISSUE THREAD

**Issue by ToxicMushroom** [kanidm/kanidm#2920]
Currently neither the Web UI or API restricts or notifies about:
- setting an **empty name**.
- setting an already **existing name**

for a TOTP entry. 

We should probably protect the users a bit here :)

**Comment by CEbbinghaus** [comment]
Is this one up for grabs?

**Comment by yaleman** [comment]
No, the API should check for a non-empty name at the very least
